src/imitation/util/video_wrapper.py

```diff
@@ -75,3 +79,69 @@ def close(self) -> None:
             self.video_recorder.close()
             self.video_recorder = None
         super().close()
+
+
+def record_and_save_video(
+    output_dir: str,
+    policy: policies.BasePolicy,
+    eval_venv: vec_env.VecEnv,
+    video_kwargs: Mapping[str, Any],
+    logger: Optional[sb_logger.Logger] = None,
+) -> None:
+    video_dir = osp.join(output_dir, "videos")
+    video_venv = VideoWrapper(
+        eval_venv,
+        directory=video_dir,
+        **video_kwargs,
+    )
+    sample_until = rollout.make_sample_until(min_timesteps=None, min_episodes=2)
+    # video.{:06}".format(VideoWrapper.episode_id) will be saved within
+    # rollout.generate_trajectories()
+    rollout.generate_trajectories(policy, video_venv, sample_until)
+    assert "video.000000.mp4" in os.listdir(video_dir)
+    video_path = osp.join(video_dir, "video.000000.mp4")
+    if logger:
+        logger.record("video", video_path)
+        logger.log(f"Recording and saving video to {video_path} ...")
+
+
+class SaveVideoCallback(callbacks.EventCallback):
+    """Saves the policy using `save_n_record_video` each time it is called.
+
+    Should be used in conjunction with `callbacks.EveryNTimesteps`
+    or another event-based trigger.
+    """
+
+    def __init__(
+        self,
+        policy_dir: str,
+        eval_venv: vec_env.VecEnv,
+        video_kwargs: Mapping[str, Any],
```

Comment:
IS this something someone will usually need to specify? If not, perhaps make it Optional and default to None?

Reply:
sounds good